sympy/polys/series/ring.py

```diff
@@ -1,109 +1,1007 @@
+
+        """
+        R = self.ring
+        series = R.cosh(s.series)
+        return self.from_element(series)
+
+
+class PowerSeriesElement(DomainElement, CantSympify, Generic[Er]):
+    """A class for representing elements of a Power Series."""
+
+    def __init__(self, ring: PowerSeriesRingRing[Er], series: TElement):
+        self.series = series
+        self.ring = ring
+
+    def __repr__(self) -> str:
+        R = self.ring.ring
+        return R.pretty(self.series, symbol=str(self.ring.symbol))
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, PowerSeriesElement):
+            return False
+
+        R = self.ring.ring
+        return R.equal_repr(self.series, other.series)
+
+    def __hash__(self) -> int:
+        return hash((self.ring, self.series))
+
+    def __pos__(self) -> PowerSeriesElement[Er]:
+        R = self.ring.ring
+        s = R.positive(self.series)
+        return self.new(s)
+
+    def __neg__(self) -> PowerSeriesElement[Er]:
+        R = self.ring.ring
+        s = R.negative(self.series)
+        return self.new(s)
+
+    def new(self, series: TElement) -> PowerSeriesElement[Er]:
+        return self.__class__(self.ring, series)
+
+    def __add__(
+        self, other: PowerSeriesElement[Er] | Er | int
+    ) -> PowerSeriesElement[Er]:
+        """
+        Add two power Series.
+
+        Examples
+        ========
+
+        >>> from sympy import QQ
+        >>> from sympy.polys.series import PowerSeriesRingField
+        >>> R = PowerSeriesRingField(QQ, "x")
+        >>> x = R.gen
+        >>> R.sin(x) + R.cos(x)
+        1 + x - 1/2*x**2 - 1/6*x**3 + 1/24*x**4 + 1/120*x**5 + O(x**6)
+
+        """
+        if not other:
+            return self
+
+        if isinstance(other, PowerSeriesElement):
+            return self._add(other)
+
+        domain = self.ring.domain
+        if isinstance(other, int):
+            return self._add_ground(self.ring.domain_new(other))
+        elif domain.of_type(other):
+            return self._add_ground(other)
+        else:
+            raise NotImplementedError
+
+    def __radd__(self, other: Er | int) -> PowerSeriesElement[Er]:
+        domain = self.ring.domain
+        if isinstance(other, int):
+            return self._add_ground(self.ring.domain_new(other))
+        elif domain.of_type(other):
+            return self._add_ground(other)
+        else:
+            raise NotImplementedError
+
+    def __sub__(
+        self, other: PowerSeriesElement[Er] | Er | int
+    ) -> PowerSeriesElement[Er]:
+        """
+        Subtract two power Series.
+
+        Examples
+        ========
+
+        >>> from sympy import QQ
+        >>> from sympy.polys.series import PowerSeriesRingField
+        >>> R = PowerSeriesRingField(QQ, "x")
+        >>> x = R.gen
+        >>> R.sin(x) - R.cos(x)
+        -1 + x + 1/2*x**2 - 1/6*x**3 - 1/24*x**4 + 1/120*x**5 + O(x**6)
+
+        """
+        if not other:
+            return self
```

Comment:
The type should be checked before doing this e.g. this is wrong:
```python
In [1]: x = QQ['x'].gens[0]

In [2]: x
Out[2]: x

In [3]: x + []
Out[3]: x
```
It should be a TypeError.